gwcs/wcs.py

```diff
@@ -501,7 +489,57 @@ def invert(self, *args, **kwargs):
         else:
             return result
 
-    def numerical_inverse(self, *args, tolerance=1e-5, maxiter=50, adaptive=True,
+    def outside_footprint(self, world_arrays):
+        world_arrays = list(world_arrays)
+
+        axes_types = set(self.output_frame.axes_type)
+        footprint = self.footprint()
+        not_numerical = False
+        if not utils.isnumerical(world_arrays[0]):
+            not_numerical = True
+            world_arrays = high_level_objects_to_values(*world_arrays, low_level_wcs=self)
+        for axtyp in axes_types:
+            ind = np.asarray((np.asarray(self.output_frame.axes_type) == axtyp))
+
+            for idim, coord in enumerate(world_arrays):
+                coord = _tofloat(coord)
+                if np.asarray(ind).sum() > 1:
+                    axis_range = footprint[:, idim]
+                else:
+                    axis_range = footprint
+                range = [axis_range.min(), axis_range.max()]
```

Comment:
This should be something like:
```python
range = sorted([wrap_ra(axis_range.min(), wrapping_angle=180 or 360), wrap_ra(axis_range.max(), wrapping_angle=180 or 360)]
```
or `_dec` (instead of `_ra`). To do this correctly, look into the transforms in `gwcs` pipeline and find CartesianToSpherical. Then look at `CartesianToSpherical.wrap_lon_at` to find the actual wrapping angle. However, this only for `lon` (`RA`). For `lat` I think the range is `[-90, 90]`. So a different treatment should be for that coordinate.

Reply:
Maybe something like this:
```python
                range = [axis_range.min(), axis_range.max()]
                if abs(range[1] - range[0]) >= 350:
                    # most likely this coordinate is wrapped at 360
                    range = [range[1] - 360, range[0]]
                elif abs(range[1] - range[0]) >= 170:
                    # most likely this coordinate is wrapped at 180
                    range = [range[1] - 180, range[0]]
```
could detect phase jumps and unwrap one of the range's limit. This may need some tweaking.

Reply:
Also, `range` is a built-in function. I suggest changing the variable name.